ote_sdk/ote_sdk/entities/datasets.py

```diff
@@ -127,6 +132,11 @@ def __init__(
         items: Optional[List[DatasetItemEntity]] = None,
         purpose: DatasetPurpose = DatasetPurpose.INFERENCE,
     ):
+        check_input_param_type(
+            OptionalParamTypeCheck(items, "items", List[DatasetItemEntity]),
```

Comment:
won't it work if  RequiredParamTypeCheck(items, "items",  Optional[List[DatasetItemEntity]]) ?

I guess there is might be a confusion, what 'optional' means

Reply:
Subscripted generics cannot be used with class and instance checks, I can try to add new branch to the check_parameter_type function

Reply:
Thanks, added InputParamTypeCheck class, removed RequiredParamTypeCheck and OptionalParamTypeCheck classes